From 074879ffcf4e68ada15b30698fcc0be7557adc47 Mon Sep 17 00:00:00 2001
From: Christopher Tate <ctate@google.com>
Date: Tue, 16 Jan 2018 17:45:44 -0800
Subject: [PATCH 057/306] Clear the right uid's alarms, please

When we clear data for an app, we need to cancel the alarms for
that app's uid -- not the *calling* uid, which could be something
like the Settings app's, i.e. the system uid.  That's bad.

Bug: 72059961
Bug: 70670726
Test: manual
Change-Id: Ib6833ce2672bc6dfab6a4be6014a0301611576e2
(cherry picked from commit 5e5c171dea2bb43013fbd72933a1b84df9dba818)
---
 .../core/java/com/android/server/am/ActivityManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 7167dcf1c3d..bf78ca43198 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -6242,7 +6242,7 @@ public class ActivityManagerService extends IActivityManager.Stub
 
                     // Clear its pending alarms
                     AlarmManagerInternal ami = LocalServices.getService(AlarmManagerInternal.class);
-                    ami.removeAlarmsForUid(uid);
+                    ami.removeAlarmsForUid(appInfo.uid);
                 }
             } catch (RemoteException e) {
             }
-- 
2.17.1

